#! /usr/bin/env python

PACKAGE='steer_controller'
import roslib
roslib.load_manifest(PACKAGE)

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gp1 = gen.add_group("GP1")
gp2 = gen.add_group("GP2")
gp3 = gen.add_group("GP3")

#        Name                  Type           Level Description     Default   Min    Max
gp1.add("ref_distance_",       double_t, 0,   "ref_distance",       2.5,      0,     5   )
gp1.add("k_angle_",            double_t, 0,   "k_angle",            1,        0,     6   )
gp1.add("k_error_",            double_t, 0,   "k_error",            3,        0,     6   )
gp2.add("Kp_error_",           double_t, 0,   "Kp_error",           0.2,      0,     2   )
gp2.add("Ki_error_",           double_t, 0,   "Ki_error",           0.2,      0,     2   )
gp3.add("filter_param_",       double_t, 0,   "filter_param",       0.6,      0,     1   )
gp3.add("Kp_wheel_",           double_t, 0,   "Kp_wheel",           1000,     0,     2000)
gen.add("update_param_",       bool_t,   0,   "update_param",       False)



exit(gen.generate(PACKAGE, "stanley_controller", "stanley_param"))
